acstools/acszpt.py

```diff
@@ -268,116 +264,97 @@ def _check_date(self, fmt='%Y-%m-%d'):
-                return
+        #check user input date, detector, and filter
+        bool_inputs = self._check_inputs()
+
+        if bool_inputs:
+
+        #select between all filters or a single filter depending on if user has
+        #specified a filter, and then generate a request body to send the API
+            if self.filt is None:
+                #URL to invoke all_filter API on AWS APIGateway
+                invokeURL = "https://vtbopx9sf3.execute-api.us-east-1." \
+                                "amazonaws.com/main/all-filter-resource"
+                #Generate a request body to send the API
+                body = '{"date": "%s", "detector": "%s"}' %(self.date, self.detector)
+            else:
+                #URL to invoke single_filter API on AWS APIGateway
+                invokeURL = "https://vtbopx9sf3.execute-api.us-east-1." \
+                                "amazonaws.com/main/single-filter-resource"
+                #Generate a request body to send the API
+                body = '{"date": "%s", "detector": "%s", "filter": "%s"}' %(self.date, self.detector, self.filt)
+
+            #send request to APIGateway with try/except clauses
+            help_desk_string = "\nIf this error persists, please contact the HST Help Desk at \n" \
+            "https://stsci.service-now.com/hst"
+
+            request_exception = True
+
+            try:
+                response = requests.post(invokeURL, data = body)
+                response.raise_for_status()
+
+            except requests.exceptions.HTTPError:
+                err_string = "HTTP Error to AWS API Gateway:" + help_desk_string
+                print(err_string)
+            except requests.exceptions.ConnectionError:
+                err_string = "Error Connecting to AWS API Gateway:" + help_desk_string
+                print(err_string)
+            except requests.exceptions.Timeout:
+                print("Timeout Error to AWS API Gateway:" + help_desk_string)
+            except requests.exceptions.RequestException:
+                print("Request Exception Error to AWS API Gateway:" + help_desk_string)
+            else: 
+                request_exception = False
+
+            if request_exception:
+                sys.exit()
+
+            #and store the results
+            APIoutput = json.loads(response.text)
+
+            #define appropriate headers for the table
+            headers = ['Detector', 'Filter', 'PHOTLAM', 'PHOTFLAM', 'STmag', 'VEGAmag', 'ABmag']
+
+            #store each row of the results
+            rows = APIoutput["rows"]
+
+            #generate an astropy QTable from the results
+            table = QTable(names = headers,
+                dtype = ('S', 'S', 'float64', 'float64', 'float64', 'float64', 'float64'))
+
+            for i in range(len(rows)):
+                table.add_row(rows[i])
+
+            #remove detector column (to match output of previous API version)
+            table.remove_column('Detector')
```

Comment:
I feel like this can be easily optimized. You know all the rows already in advance. And why even add Detector column only to remove it later? See https://docs.astropy.org/en/latest/table/index.html#performance-tips

Reply:
the output of the API includes the detector line in the first position (because that's how it shows on the webpage), but we are removing it here in the python module to keep consistent with the output from the older version